Modules/Core/Common/include/itkMakeFilled.h

```diff
@@ -0,0 +1,47 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkMakeFilled_h
+#define itkMakeFilled_h
+
+namespace itk
+{
+/** Yields a container of the specified type, filled by the specified value, assigned to each of its elements.
+ * Only intended for container types of a compile-time fixed size, including instances of FixedArray (and derived
+ * classes), Index, Offset, Size, and std::array. Moreover, their size (number of elements) should be greater than zero.
+ */
+template <typename TContainer>
+constexpr TContainer
+MakeFilled(typename TContainer::const_reference value)
+{
+  // Note: The seemingly redundant {} initialization is required for C++14, in order to declare MakeFilled constexpr.
+  TContainer container{};
```

Comment:
Perhaps we could provide hints to future C++17 or C++20 code maintainters

```c++
#if __cplusplus >= 201703L
TContainer container;
#else
TContainer container{};
#endif
```

At some point, ITK will require C++17 or C++20, and it will help future developers to know these details.  This is certainly not necessary, but if you already know what the future code should look like, now might be the time to document that.

Reply:
Thanks for the suggestion @hjmjohnson However, I hesitate, because in general, I would rather avoid `#if` directives. And specifically in this case, I'm pretty sure that the redundant `{}` initialization will be optimized away with any Release build.

Would it not be sufficient that the comment explicitly mentions "C++14"? So that it can be easily found by a global search for "C++14", when upgrading to C++17?

For the sake of completeness, I just tried:

```
constexpr int Answer() {
    int uninitializedVariable;
    uninitializedVariable = 42;
    return uninitializedVariable;
}

```

At https://godbolt.org/z/an8hhbEeK  Interestingly, it says (clang 13.0.1):

> warning: uninitialized variable in a constexpr function is a C++20 extension [-Wc++20-extensions]

So it may still take a while before we can say goodbye to those `{}` !